### PR TITLE
wasabibackend: 1.1.12 -> 1.1.13.0, use buildDotnetModule

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
#146257

This does seem to crash when launching the executable, though that happens on master as well.
```shell
~/nix/nixpkgs > ./result/bin/WasabiBackend
2021-12-31 01:44:00 INFO        InitConfigStartupTask (35)      Wasabi Backend started (38c871a8-8a55-4722-b340-92da8fce08bd).
2021-12-31 01:44:00 INFO        InitConfigStartupTask (42)      Config is successfully initialized.
2021-12-31 01:44:01 INFO        InitConfigStartupTask (47)      RoundConfig is successfully initialized.
2021-12-31 01:44:01 CRITICAL    Program (37)    System.Net.Http.HttpRequestException: Connection refused
 ---> System.Net.Sockets.SocketException (111): Connection refused
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at NBitcoin.RPC.RPCClient.SendCommandAsyncCore(RPCRequest request, Boolean throwIfRPCError)
   at NBitcoin.RPC.RPCClient.SendCommandAsync(RPCRequest request, Boolean throwIfRPCError)
   at NBitcoin.RPC.RPCClient.GetBlockchainInfoAsync()
   at WalletWasabi.BitcoinCore.RpcClientBase.GetBlockchainInfoAsync()
   at WalletWasabi.Backend.Global.AssertRpcNodeFullyInitializedAsync()
   at WalletWasabi.Backend.Global.InitializeAsync(Config config, CoordinatorRoundConfig roundConfig, IRPCClient rpc, CancellationToken cancel)
   at WalletWasabi.Backend.InitConfigStartupTask.ExecuteAsync(CancellationToken cancellationToken)
   at WalletWasabi.Backend.Extensions.RunWithTasksAsync(IHost host, CancellationToken cancellationToken)
   at WalletWasabi.Backend.Program.Main(String[] args)
``` 

You can now fetch the nuget dependencies using `nix-build -A wasabibackend.passthru.fetch-deps`.

cc @mmahut @prusnak @Stunkymonkey

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/wasabibackend/default.nix [hunk 1]
```diff
@@ -1,95 +1,45 @@
...
-  ];
+  projectFile = "WalletWasabi.Backend/WalletWasabi.Backend.csproj";
+  nugetDeps = ./deps.nix;
 
-  buildPhase = ''
-    export HOME=$(mktemp -d)
-    export DOTNET_CLI_TELEMETRY_OPTOUT=1
-    export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
-    export DOTNET_ROOT="${dotnet-sdk}/bin"
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
```
Comment (prusnak) [inline pkgs/applications/blockchains/wasabibackend/default.nix:25]: Shouldn't this be `dotnet-sdk = dotnetCorePackages.sdk_3_1;` ?

Comment (IvarWithoutBones) [reply pkgs/applications/blockchains/wasabibackend/default.nix:25]: Their readme suggests using SDK 6, i think this is just another instance of a mismatching SDK and runtime. Happens in other packages as well.

Comment (prusnak) [reply pkgs/applications/blockchains/wasabibackend/default.nix:25]: Their master is already at Wasabi 2.x which is fundamentally different to what we are packaging here (v 1.x).

Readme from the `v1.1.13.0` tag mentions SDK 3.1: https://github.com/zkSNACKs/WalletWasabi/blob/v1.1.13.0/README.md

Comment (IvarWithoutBones) [reply pkgs/applications/blockchains/wasabibackend/default.nix:25]: Ah, I forgot to switch to that tag. My bad, should be fixed now!

**Comment by prusnak** [comment]
@GrahamcOfBorg build wasabibackend

**Comment by IvarWithoutBones** [comment]
Thanks for the review :)

**Pull request merged**